### ISSUE TITLE
To Do List

### ISSUE THREAD

**Issue by luckynozomi** [suixin661014/Project-5168#1]
- [ ] Add support for local response layers
- [x] Optimize for GPU performance
- [x] Export and Import training results
- [ ] Parameter Selection in CNN
- [ ] Write logistic regression as a subclass of FCLayer
- [ ] Make the code more easy to use, no need to modify the code to reconstruct the layer structure
- [x] Data augmentation in training
- [ ] Force every kernel to learn independent feature
- [ ] Selection of activation function and loss function
- [x] Subset the data to get a dataset to experiment on
- [ ] Learn soft weight sharing and find paper on the performance
- [ ] Learn early stopping and understand more complicated gradient descent algorithms

**Comment by luckynozomi** [comment]
- Data augmentation in training is implemented.
  - Reflection
  - Random shifting

**Comment by luckynozomi** [comment]
-  GPU performance optimized for cuDNN
  - use FFT for faster convolution
  - code optimized when needs padding
- Export and Import training results implemented
- No need to subset the data to experiment on
